File: pichub.client/src/entities/ChatLineDTO.ts
export interface ChatLineDTO {
  id: number;
  privateChatId: string | null;
  groupChatId: string | null;
  senderId: string;
  replyingToId: number | null;
  content: string;
  createdAt: Date;
  lastUpdatedAt: Date | null;
}

File: PicHub.Server/DTOs/ChatLineDTO.cs
using Newtonsoft.Json;

namespace PicHub.Server.DTOs
{
    public class ChatLineDTO
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("privateChatId")]
        public string? PrivateChatId { get; set; }

        [JsonProperty("groupChatId")]
        public string? GroupChatId { get; set; }

        [JsonProperty("senderId")]
        public required string SenderId { get; set; }

        [JsonProperty("replyingToId")]
        public int? ReplyingToId { get; set; }

        [JsonProperty("content")]
        public required string Content { get; set; }

        [JsonProperty("createdAt")]
        public DateTime CreatedAt { get; set; }

        [JsonProperty("lastUpdatedAt")]
        public DateTime? LastUpdatedAt { get; set; }
    }
}
